packages/core/admin/server/controllers/api-token.js

```diff
@@ -38,6 +38,21 @@ module.exports = {
     ctx.created({ data: apiToken });
   },
 
+  async regenerate(ctx) {
+    const { id } = ctx.params;
+    const apiTokenService = getService('api-token');
+
+    const apiTokenExists = await apiTokenService.getById(id);
+    if (!apiTokenExists) {
```

Comment:
What if the token is expired? Should we still regenerate it?

Reply:
Expiration doesn't exist at this point, that will be covered in a PR to be created sometime this afternoon. But to answer the question: yes, I think regenerate should still work, it just won't do anything particularly useful. It won't reset the lifespan of the token, it will regenerate an accessKey for the expired token. Those two ideas should be independent of each other.

To get a head start on something I think is related to what you're asking, I think an expired token should be able to have its lifespan renewed (though probably a separate feature to be added later), but I don't believe extending the lifespan should affect the accessKey. Changing the accessKey will invalidate whatever is currently using it, and in my opinion the point of having/extending the lifespan of a token is to prevent having to unnecessarily replace accessKeys or delete/recreate tokens.